pdh_windows.go

```diff
@@ -0,0 +1,120 @@
+package sigar
+
+/* These are convenience methods for extracting metrics from the Windows
+   Performance Data Helper (PDH) library */
+
+import (
+	"fmt"
+	"github.com/scalingdata/win"
+	"unsafe"
+)
+
+/* Run the set of provided queries once and report back the raw counter values for each.
+   Queries must have a wildcard, like `\Processor(*)\% idle time`. Returns a map of
+   wildcard values to arrays of results. The indices in the array match up with the
+   index of the query in `queries`. */
+func runRawPdhArrayQueries(queries []string) (map[string][]uint64, error) {
+	var query win.PDH_HQUERY
+	counters := make([]win.PDH_HCOUNTER, len(queries))
+	results := make(map[string][]uint64, len(queries))
+	// Open a new query
+	err := win.PdhOpenQuery(0, 0, &query)
+	if err != 0 {
+		return nil, fmt.Errorf("Error opening PDH query: %X", err)
```

Comment:
There seems to be a pattern in this PR of wrapping errors instead of straight returning them.  Is this a style from this repo?